File: front/income-flow/src/app/components/income/income.ts
import {Component, signal, inject} from '@angular/core';
import {MatButtonModule} from '@angular/material/button';
import { form, Field } from '@angular/forms/signals'
import {IncomeService} from './income.service';

@Component({
  selector: 'app-income',
  imports: [MatButtonModule, Field],
  templateUrl: './income.html',
  styleUrl: './income.scss',
})
export class Income {
  incomeForm = signal({
    goods_id: 0,
    section_id: 0,
    goods_count: 0,
    contractors_id: 0
  })
  form = form(this.incomeForm)
  readonly incomeService = inject(IncomeService);

  income() {
    const form = this.incomeForm();

    this.incomeService.income(form.goods_id, form.goods_count, form.contractors_id, form.section_id)
      .subscribe(income => {
        console.log(income)
      })
  }
}

File: back/income_rest_api/internal/model/income.go
package model

type Income struct {
	ID            int `json:"id" db:"id"`
	GoodsID       int `json:"goods_id" db:"goods_id"`
	SectionID     int `json:"section_id" db:"section_id"`
	GoodsCount    int `json:"goods_count" db:"goods_count"`
	ContractorsID int `json:"contractors_id" db:"contractors_id"`
}
